activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -469,7 +480,10 @@ def find_one(id)
           MSG
         end
 
-        relation = where(primary_key => id)
+        pk_columns = Array(primary_key)
+        pk_values = Array(id)
+
+        relation = where(pk_columns.zip(pk_values).to_h)
```

Comment:
This, on the other hand, feels avoidably-allocatey for something that's worth keeping tight. Pending a `where([] => ..)` option, my instinct is that this is probably worth branching on `primary_key.is_a?(Array)`.

Reply:
Actually that's fair, I totally agree with the argument that having a branch for this context clearly reveals an option to get rid of the branch if we ever support `where([] => ..)` syntax. I'll add a branch